### PR TITLE
Feature/sdl 0269 new vehicle data climate data

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
Implements https://github.com/smartdevicelink/sdl_core/issues/3198

This PR is **[ready]** for review.

### Risk
This PR makes **[minor]** API changes.

### Testing Plan
ATF test scripts from https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2491

### Summary
Add new vehicle data ClimateData
Use rpc spec from https://github.com/smartdevicelink/rpc_spec/pull/299

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by KhrystynaDubovyk** [comment]
@atiwari9 please review this PR

File: .gitmodules [hunk 1]
```diff
@@ -3,4 +3,5 @@
 	url = https://github.com/smartdevicelink/jsoncpp.git
 [submodule "tools/rpc_spec"]
 	path = tools/rpc_spec
-	url = https://github.com/smartdevicelink/rpc_spec.git
+	url = https://github.com/LuxoftSDL/rpc_spec.git
```
Comment (atiwari9) [inline .gitmodules:6]: Temporary commit? Should be removed for donation.

**Review by atiwari9 — APPROVED** [review]
Approved contingent on successful test pass of test scripts: https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2491

**Comment by LitvinenkoIra** [comment]
@JackLivio This PR is ready for Livio review. Thank you!

**Review by ShobhitAd — CHANGES_REQUESTED** [review]

File: src/appMain/sdl_preloaded_pt.json [hunk 2]
```diff
@@ -3475,13 +3479,44 @@
                     "maxsize": 100,
                     "since": "5.0"
                 },
+                {
+                    "name": "climateData",
+                    "params": [
+                        {
+                            "name": "externalTemperature",
+                            "key": "OEM_REF_EXT_TEMP",
+                            "type": "Temperature",
```
[the remark this answers is not in the document]
Comment (LitvinenkoIra) [reply src/appMain/sdl_preloaded_pt.json]: @ShobhitAd Thank you! Done in https://github.com/smartdevicelink/sdl_core/pull/3586/commits/3dc17fa5cf0955dbeefd8db7f2a7b2f4e789fbd8

File: src/appMain/sdl_preloaded_pt.json [hunk 3]
```diff
@@ -3475,13 +3479,44 @@
...
+                            "mandatory": false
+                        },
+                        {
+                            "name": "cabinTemperature",
+                            "key": "OEM_REF_CABIN_TEMP",
+                            "type": "Temperature",
```
[the remark this answers is not in the document]
Comment (LitvinenkoIra) [reply src/appMain/sdl_preloaded_pt.json]: @ShobhitAd Thank you! Done in https://github.com/smartdevicelink/sdl_core/pull/3586/commits/3dc17fa5cf0955dbeefd8db7f2a7b2f4e789fbd8

File: src/appMain/sdl_preloaded_pt.json [hunk 4]
```diff
@@ -3475,13 +3479,72 @@
...
+                            "type": "Struct",
+                            "mandatory": false
+                        },
+                        {
+                            "name": "cabinTemperature",
+                            "key": "OEM_REF_CABIN_TEMP",
+                            "params": [
+                                {
+                                    "name": "unit",
+                                    "key": "OEM_REF_EXT_TEMP_UNIT",
```
Comment (ShobhitAd) [inline src/appMain/sdl_preloaded_pt.json:3511]:
```suggestion
                                    "key": "OEM_REF_CABIN_TEMP_UNIT",
```

File: src/appMain/sdl_preloaded_pt.json [hunk 5]
```diff
@@ -3475,13 +3479,72 @@
...
+                                    "type": "TemperatureUnit",
+                                    "mandatory": true
+                                },
+                                {
+                                    "name": "value",
+                                    "key": "OEM_REF_EXT_TEMP_VALUE",
```
Comment (ShobhitAd) [inline src/appMain/sdl_preloaded_pt.json:3517]:
```suggestion
                                    "key": "OEM_REF_EXT_TEMP_VALUE",
```
```suggestion
                                    "key": "OEM_REF_CABIN_TEMP_VALUE",
```

**Review by ShobhitAd — CHANGES_REQUESTED** [review]

File: src/appMain/sdl_preloaded_pt.json (hunk 4 above)
Comment (LitvinenkoIra) [reply src/appMain/sdl_preloaded_pt.json]: Done in 13364d006fbe906bff31e55c50fc65457d881c76

File: src/appMain/sdl_preloaded_pt.json (hunk 5 above)
Comment (LitvinenkoIra) [reply src/appMain/sdl_preloaded_pt.json]: Done in 13364d006fbe906bff31e55c50fc65457d881c76

**Review by ShobhitAd — APPROVED** [review]

**Comment by ShobhitAd** [comment]
@LitvinenkoIra please update the `tools/rpc_spec` submodule commit to https://github.com/smartdevicelink/rpc_spec/pull/299/commits/bdbdaf4fc14742361f4eb0a858ed1fe79834c692

**Comment by LitvinenkoIra** [comment]
> @LitvinenkoIra please update the `tools/rpc_spec` submodule commit to [smartdevicelink/rpc_spec@bdbdaf4](https://github.com/smartdevicelink/rpc_spec/commit/bdbdaf4fc14742361f4eb0a858ed1fe79834c692)

Done in c6e25a2

**Comment by ShobhitAd** [comment]
@LitvinenkoIra Please resolve merge conflicts and update the rpc spec submodule commit

**Pull request merged**